module/vision/BallDetector/src/BallDetector.cpp

```diff
@@ -106,137 +108,168 @@ namespace module::vision {
+                        radius = uBCw.dot(uECw) < radius ? uBCw.dot(uECw) : radius;
                     }
 
-                    // Ball cam space info
-                    b.cone.axis   = horizon.Hcw.topLeftCorner<3, 3>().cast<float>() * axis;
-                    b.cone.radius = radius;
+                    // The vectors are in world space, multiply by Rcw to get the central axis in camera space
+                    b.uBCc   = Hcw.rotation() * uBCw;
+                    b.radius = radius;  // arccos(radius) is the angle between the central axis and edge of the ball
+
+                    // CALCULATE DISTANCE TO BALL WITH TWO METHODS
+                    // 1. Angular-based distance
+                    // 2. Projection-based distance
 
+                    // Angular-based distance
                     // https://en.wikipedia.org/wiki/Angular_diameter
+                    // From the link, the formula with arcsin is used since a ball is a spherical object
+                    // The variables are:
+                    //      delta: 2 * arccos(radius)
+                    //      d_act: 2 * field.ball_radius
+                    //      D: distance
+                    // Rearranging the equation gives
+                    //      distance = (field.ball_radius) / (sin(arccos(radius)))
+                    // Using sin(arccos(x)) = sqrt(1 - x^2)
+                    //      distance = field.ball_radius / sqrt(1 - radius^2)
                     float distance = field.ball_radius / std::sqrt(1.0f - radius * radius);
+                    // Convert uBCc into a position vector (rBCc) and then convert it into Spherical Reciprocal
+                    // Coordinates (1/distance, phi, theta)
+                    b.srBCc              = cartesianToReciprocalSpherical(b.uBCc * distance);
+                    b.screen_angular_BCw = cartesianToSpherical(uBCw).tail<2>();  // no depth (x-coordinate)
+
+                    // Projection-based distance
+                    // Given a flat X-Y plane that intersects through the middle of the ball, find the point where
+                    // the central axis vector (uBCw) intersects with this place. Get the distance from this point.
+                    // https://en.wikipedia.org/wiki/Line%E2%80%93plane_intersection#Algebraic_form
+                    //      Plane normal: (0, 0, 1)
+                    //      Point on plane: (0, 0, rWCw.z() + field.ball_radius)
+                    //      Line direction: uBCw
+                    //      Point on line: (0, 0, 0)
+                    // Since the plane normal zeros out x and y, only consider z
+                    // rWCw = -Hcw.inverse().translation().z()
+                    const float d                   = (field.ball_radius - Hcw.inverse().translation().z()) / uBCw.z();
+                    const Eigen::Vector3f srBCc     = uBCw * d;  // used later, so save it in a variable
```

Comment:
I don't get why this is `srBCc`. `uBCw` is just a normal unit vector to the ball, then scaled by `d`, the distance, to get `rBCw`?
And below, the norm would just be `d` right? Why make two variables for the same thing?

Reply:
It should be `rBCc`, not `srBCc` ... not sure why we made the extra variable though, maybe a remnant of some old code that got deleted.